src/olympia/api/jwt_auth/views.py

```diff
@@ -24,6 +26,34 @@ class JWTKeyAuthentication(JSONWebTokenAuthentication):
     by the authenticated user.
     """
 
+    def authenticate(self, request):
+        """
+        Returns a two-tuple of `User` and token if a valid signature has been
+        supplied using JWT-based authentication.  Otherwise returns `None`.
+
+        Copied from rest_framework_jwt BaseJSONWebTokenAuthentication, with
+        the decode_handler changed to our own - because we don't want that
+        decoder to be the default one in settings.
+        """
+        jwt_value = self.get_jwt_value(request)
+        if jwt_value is None:
+            return None
+
+        try:
+            payload = handlers.jwt_decode_handler(jwt_value)
+        except jwt.ExpiredSignature:
```

Comment:
aha, now that you've moved this code under our control, you can add logging! I'd like to see this actual exception message logged. I've had to go pdb spelunking to find it before :(